### PR TITLE
Add basic demo mode just for admins

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: lib-classifier, lib-grommet-theme

Describe your changes:
This adds the gear menu and option toggle on demo mode if you are an admin user. This can be tested with the project app and with an admin user. There will be a gear only visible to an admin user in the task navigation button area where it can be enabled:

demo-mode

Check it out by bootstrapping and running `yarn dev` with the project app and go to: http://local.zooniverse.org:3000/projects/brooke/i-fancy-cats/classify

and sign in with an admin user.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
In the future, to get to parity with PFE classifier function, this needs to be expanded to project owners and collaborators.

File: packages/lib-classifier/src/store/ClassificationStore.spec.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/ClassificationStore.spec.js:292]: Maybe spy here too, just in case there is a bug in `submitClassification`, which we'll miss by not running the action.

File: packages/lib-classifier/src/store/WorkflowStepStore.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/WorkflowStepStore.js:239]: Making a note for myself that this fix can be removed from #1859.

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js:86]: Stray console log.

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js:91]: Does this test still need to be fixed?

**Review by eatyourgreens — APPROVED** [review]
Looks good to me. I have minor questions but nothing blocking.

Logged in, I was able to toggle demo mode with the trackpad and stop classifications from being submitted. I could also operate the toggle from the keyboard, but it took some figuring out and should be improved in a future PR. For keyboard users, I'd recommend cutting down on the number of keystrokes retired to navigate to the toggle and operate it. Ideally, after opening the toggle, I think a single keystroke should operate the checkbox. At the moment, you have to tab to it before you can operate it. Maybe autofocus the checkbox after the popup opens?

After the popup is closed (with Esc), focus is lost and it takes a while to figure out where it's gone. I'd recommend returning focus to the cog button after the toggle is closed from the keyboard. All this was tested from the keyboard in Firefox 82.

Screenshot of the dev classifier showing the deno mode toggle next to the Done button for I Fancy Cats.

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.js:29]: Sorry, missed this on my first pass. I think useEffect is supposed to return a callback function.
```suggestion
    return () => setIsAdmin(false)
```

Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.js:29]: My suggestion could also be a named function, to make debugging easier.
```js
return function onUnmount() {
  setIsAdmin(false)
}
```

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js:91]: For reference, this test is skipped because it throws an error inside the `DropButton` that's rendered for the toggle.

```
TypeError: Cannot read property 'length' of undefined
    at phash (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:683:13)
    at ComponentStyle.generateAndInjectStyles (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1271:25)
    at useInjectedStyle (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1497:120)
    at useStyledComponentImpl (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1525:28)
    at StyledButton (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/styled-components/dist/styled-components.cjs.js:1592:12)
```

My best guess at what might be wrong here is that `styled-components` uses context, but this test is mocking `React.useContext`.

Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/ExpertOptions/ExpertOptionsContainer.spec.js:91]: One way to get around this would be to add an error boundary around the `DropButton`. It's not ideal, but I think it would allow you to test the container.

File: packages/lib-classifier/src/store/ClassificationStore.spec.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/ClassificationStore.spec.js:273]: With demo mode enabled, we still expect the old classification to complete and a new classification to be created. Is it possible to test that `classifications.active` changes?

**Comment by eatyourgreens** [comment]
I'm seeing the current user checked on every step of I Fancy Cats, because the Expert Options component is unmounted and mounted on each change to the task navigation. That might cause us performance problems on popular projects. The user could be stored (or passed down from the project as a prop) to save on redundant API calls.

Screenshot of the dev console for I Fancy Cats, showing three API calls to check the current user during a single classification.

**Comment by eatyourgreens** [comment]
That API performance problem might need to be fixed before this can be used in production, so maybe hold off on merging until we know for certain whether it's a major issue?

**Comment by srallen** [comment]
There's enough comments here that any follow up should necessitate another review, so I'm going to dismiss the approval so this doesn't accidentally merge.

**Comment by srallen** [comment]
I came across a note I left myself about implementing this:

> We could implement this as a URL query param rather than linked to authenticated user roles

Which is still a good idea. There isn't exactly a reason that this can't be available to anyone as long as it's clearly communicated in the UI when it's enabled. So would a refactor to make the gear and toggle available in the UI by query param in the URL be welcome? Then I can eliminate the request to check for admin user (and the later requirement to check for user roles)

**Comment by eatyourgreens** [comment]
The query param sounds like a good idea to me. In Next 9.3 and up, you can get query params from the request in `getServerSideProps()`. #1826 is blocking us from using that in production at the moment, because it will make requests to `https://www.zooniverse.org/_next/data/`.

**Comment by beckyrother** [comment]
Small design comment that the flyout in dark mode needs to be [darker gray](https://projects.invisionapp.com/dsm/zooniverse/primary-brand/asset/colors/5bfd7a8a86b10b0011547f68) to stand out from the rest of the page.

**Comment by srallen** [comment]
Closing this in favor of #1909. I think I addressed all the comments that still applied.

**Pull request closed without merging**
